### PR TITLE
BUG: series sum changes dtype uint64 to int64

### LINKED ISSUE #53401: BUG: Pandas uint64 sums to int64
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import numpy as np
import pandas as pd

s = pd.Series([10000000000000000000], dtype="uint64")
s.sum()  # -8446744073709551616
s.to_numpy().sum()  # 10000000000000000000

s = pd.Series([10000000000000000000], dtype="uint64[pyarrow]")
s.sum()  # 10000000000000000000
```

### Issue Description

The summation of a series of `uint64` is converted to `int64`.

### Expected Behavior

The result would be a uint64.

### Installed Versions

<details>

``` yaml

INSTALLED VERSIONS
------------------
commit           : 37ea63d540fd27274cad6585082c91b1283f963d
python           : 3.11.3.final.0
python-bits      : 64
OS               : Linux
OS-release       : 6.2.6-76060206-generic
Version          : #202303130630~1683753207~22.04~77c1465 SMP PREEMPT_DYNAMIC Wed M
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 2.0.1
numpy            : 1.24.3
pytz             : 2023.3
dateutil         : 2.8.2
setuptools       : 67.7.2
pip              : 23.1.2
Cython           : None
pytest           : 7.3.1
hypothesis       : None
sphinx           : 5.3.0
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.9.2
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.1.2
IPython          : 8.13.2
pandas_datareader: None
bs4              : 4.12.2
bottleneck       : None
brotli           : 
fastparquet      : 2023.4.0
fsspec           : 2023.5.0
gcsfs            : None
matplotlib       : 3.7.1
numba            : 0.57.0
numexpr          : None
odfpy            : None
openpyxl         : 3.1.2
pandas_gbq       : None
pyarrow          : 11.0.0
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.10.1
snappy           : 
sqlalchemy       : 1.4.46
tables           : None
tabulate         : 0.9.0
xarray           : 2023.5.0
xlrd             : None
zstandard        : None
tzdata           : 2023.3
qtpy             : None
pyqt5            : None
```

</details>

### PR DESCRIPTION
**Pull request by srkds**
- [X] closes #53401 
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

🚫Before
image

✅After changes
image

### CODE REVIEW

**Comment by topper-123** [comment]
Thanks, @srkds. 

This should be handled in `_get_dtype_max`, unless I'm missing something . You should also add tests for this and an entry in the v2.1 whatsnew.

**Comment by srkds** [comment]
I missed the `_get_dtype_max` method while checking the execution flow. Thanks for the suggestion 👍
Added test case and entry in `whatsnew` file.

Thanks!

File: pandas/tests/series/methods/test_sum.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import pandas._testing as tm
+
+
+class TestSeriesSum:
+    def test_sum_uint64(self):
+        # GH 53401
+        s = Series([10000000000000000000], dtype="uint64")
+        result = s.sum()
+        expected = np.uint64(10000000000000000000)
+        tm.assert_almost_equal(result, expected)
```
Comment (topper-123) [inline pandas/tests/series/methods/test_sum.py:13]: Can you move the test over to `pandas.tests.reductions.test_reductions.TestSeriesReductions`.

**Review by topper-123 — COMMENTED** [review]
Generally looks good, just a single comment.

File: pandas/tests/series/methods/test_sum.py (hunk 1 above)
Comment (srkds) [reply pandas/tests/series/methods/test_sum.py:13]: yes! That seems right place.

**Review by topper-123 — APPROVED** [review]
LGTM.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @srkds

**Comment by topper-123** [comment]
Nice one, @srkds.

**Comment by srkds** [comment]
Thanks @topper-123 and @mroeschke for reviewing and merging.
I have made progress from contributing to docs fixing typos, then adding test cases, and now contributed a bug fix 🎉😀.
It wouldn't be possible without maintainers like you.
Looking forward to contributing more ❤.

**Pull request merged**